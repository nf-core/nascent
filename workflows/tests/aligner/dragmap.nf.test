nextflow_pipeline {

    name "DRAGMAP"
    script "../../../main.nf"

    test("Should run with defaults") {

        when {
            params {
                outdir = "$outputDir"
                aligner = "dragmap"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml")).match("software_versions") },
                { assert snapshot(
                    workflow.trace.tasks().size(),
                    UTILS.getAllFilesFromDir("$outputDir/preprocessing/fastp/", ".json"),
                    bam("$outputDir/dragmap/cd4_REP1.sorted.bam").getSamLinesMD5(),
                    bam("$outputDir/dragmap/cd4_REP2.sorted.bam").getSamLinesMD5(),
                    bam("$outputDir/dragmap/jurkat.sorted.bam").getSamLinesMD5(),
                    // NOTE Not deterministic
                    // path("$outputDir/dragmap/log").list(),
                    path("$outputDir/dragmap/samtools_stats").list(),
                    path("$outputDir/quality_control/bbsplit").list(),
                    UTILS.getAllFilesFromDir("$outputDir/quality_control/preseq", ".txt"),
                    path("$outputDir/quality_control/rseqc/read_duplication/xls").list(),
                    path("$outputDir/quality_control/rseqc/read_duplication/rscript").list(),
                    path("$outputDir/quality_control/rseqc/infer_experiment").list(),
                    path("$outputDir/quality_control/rseqc/read_distribution").list(),
                    path("$outputDir/coverage_graphs/").list(),
                    path("$outputDir/transcript_identification/homer/cd4.bed"),
                    path("$outputDir/transcript_identification/homer/jurkat.bed"),
                    path("$outputDir/transcript_identification/intersect/").list(),
                    path("$outputDir/transcript_identification/filtered/").list(),
                    path("$outputDir/multiqc/multiqc_report.html").exists(),
                ).match("output_files")
                }
            )
        }
    }
}
