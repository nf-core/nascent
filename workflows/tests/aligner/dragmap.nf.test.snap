{
    "output_files": {
        "content": [
            77,
            [
                "cd4_REP1.trimmed.fastp.json:md5,bd9a3344c1591d6be4d524451f9dca53",
                "cd4_REP2.trimmed.fastp.json:md5,bc6e3f9ff7835f220535cc393b8eb25f",
                "jurkat.trimmed.fastp.json:md5,65f7247a87479c7157c7357d575336e2"
            ],
            "9294458e8828aa4446fc6ea24f828d11",
            "767cae9340255d914b40f5f0a76f9d65",
            "b47414b7dada10ad11535d36af2e2cb",
            [
                "cd4_REP1.sorted.bam.flagstat:md5,6a27aa41830e5f10a2c54c8a05611f9a",
                "cd4_REP1.sorted.bam.idxstats:md5,75eaf648cea73812913344af3851ce16",
                "cd4_REP1.sorted.bam.stats:md5,80bfe47bc5be703b51313a4b2fbcc57f",
                "cd4_REP2.sorted.bam.flagstat:md5,0355e87e727027ee381053c49123336f",
                "cd4_REP2.sorted.bam.idxstats:md5,4a68a3a04440acdced8802c29ffc634c",
                "cd4_REP2.sorted.bam.stats:md5,24a85ae13a01fdbd1e6e882cf3ef55e0",
                "jurkat.sorted.bam.flagstat:md5,2ac2d8fdb09b0bf5061d9e24501c41c3",
                "jurkat.sorted.bam.idxstats:md5,8aebada21b179c96f888d08bb4ae1651",
                "jurkat.sorted.bam.stats:md5,104f00b802be80cc252beb0d4cfd709b"
            ],
            [
                "cd4_REP1.coverage.hist.txt:md5,bff6862bc8e2015eef927b40671dec1b",
                "cd4_REP1.coverage.stats.txt:md5,c1e912cd0bebc51b289160ff7ab3365a",
                "cd4_REP2.coverage.hist.txt:md5,edadf440d1c3658118ffd6b6a46821f8",
                "cd4_REP2.coverage.stats.txt:md5,cb1ce057d05fc53e7ad9b0b2e3d5aaad",
                "jurkat.coverage.hist.txt:md5,468c864f655947edb3b306cb7a72633d",
                "jurkat.coverage.stats.txt:md5,338f3095395c8d8f172e0e27f9a5a60a"
            ],
            [
                "cd4_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP1.lc_extrap.txt:md5,a2dfb0a61cd0ada1f306aaf8c7cc7b8f",
                "cd4_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP2.lc_extrap.txt:md5,f5bc717609e44a4e94a1b1c626c1aaad",
                "jurkat.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat.lc_extrap.txt:md5,129145e5b3fa3cd8a6908bcbe75ea010"
            ],
            [
                "cd4_REP1.pos.DupRate.xls:md5,c53ebd47cad3f3524dafd3d0bdd2cc71",
                "cd4_REP1.seq.DupRate.xls:md5,6fc39432eb1815a1aebac84c5c46aa97",
                "cd4_REP2.pos.DupRate.xls:md5,96b7100c85fd83d7634ebcd199de2aea",
                "cd4_REP2.seq.DupRate.xls:md5,dac033fb9e548d1e23069d9f76f9e67d",
                "jurkat.pos.DupRate.xls:md5,3389108003e203614790dd835c84bd81",
                "jurkat.seq.DupRate.xls:md5,0950602857db220cdf550f66e0ad69b8"
            ],
            [
                "cd4_REP1.DupRate_plot.r:md5,d04a5d1a2c83c2be6dd373710d63fecf",
                "cd4_REP2.DupRate_plot.r:md5,18afd2c9eb78d75dab9ac483b38111f8",
                "jurkat.DupRate_plot.r:md5,2265434abacdb7130dd1643114482c8c"
            ],
            [
                "cd4_REP1.infer_experiment.txt:md5,e257827c09d08e0211e6b55b78c09817",
                "cd4_REP2.infer_experiment.txt:md5,cfc76745a36e9144eb03b85abb40a2bc",
                "jurkat.infer_experiment.txt:md5,14af4229684e813974e4c4ac5724bd81"
            ],
            [
                "cd4_REP1.read_distribution.txt:md5,82bb213b23ea5a2c719c09ed993b576d",
                "cd4_REP2.read_distribution.txt:md5,139a91e2f36b9c93b09fe36aff3414cd",
                "jurkat.read_distribution.txt:md5,982e7cb76ca5978ff8c8bc63d17a1fbe"
            ],
            [
                "cd4_REP1.dreg.bedGraph:md5,aefd2accdaf19635f5a566e6e36a578e",
                "cd4_REP1.minus.bedGraph:md5,d4ddabe2599211e89c0eeb07544a2bf2",
                "cd4_REP1.minus.bigWig:md5,7f0614afba6a3f37d507d1ee0ce4de2f",
                "cd4_REP1.plus.bedGraph:md5,2ebf9f9cf845e8403cceda70e254be03",
                "cd4_REP1.plus.bigWig:md5,de63345606909c31c3dfa2a0d57e68bc",
                "cd4_REP2.dreg.bedGraph:md5,68cb12ce50649175d359aa8d7cbaae06",
                "cd4_REP2.minus.bedGraph:md5,96876aaaca35003be86a25c78733c8f9",
                "cd4_REP2.minus.bigWig:md5,228ed1c4fef63675c260c4f750b7c8f3",
                "cd4_REP2.plus.bedGraph:md5,a6f04b8484428786a3e089ea5509240b",
                "cd4_REP2.plus.bigWig:md5,cfea092121f880010f3ae652025e446b",
                "jurkat.dreg.bedGraph:md5,337d974e56a047cea3ce1aa2a694b1ac",
                "jurkat.minus.bedGraph:md5,c57d91865736e23f13660f9d7ecb6e5d",
                "jurkat.minus.bigWig:md5,22daac83d393dcca8f1cc96580d3d4f3",
                "jurkat.plus.bedGraph:md5,ddabf19e2172df4b80b7ce434035403e",
                "jurkat.plus.bigWig:md5,ca73f44638c066f25261efffa1eedc7f"
            ],
            "cd4.bed:md5,96b0e0bf77f3a04efcbfaf0fcdae926a",
            "jurkat.bed:md5,31454804c53975171a763d3049406598",
            [
                "cd4_intersect.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                "jurkat_intersect.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                "cd4_filtered.bed:md5,063a59e4a2612e8eedaa443d02f6d271",
                "jurkat_filtered.bed:md5,6625c2c4f2bbbf9ff2c93e587990b906"
            ],
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-11T20:13:30.191639621"
    },
    "software_versions": {
        "content": [
            {
                "BBMAP_PILEUP": {
                    "bbmap": 39.01,
                    "samtools": "1.16.1",
                    "pigz": 2.6
                },
                "BEDTOOLS_GENOMECOV_MINUS": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_GENOMECOV_PLUS": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_INTERSECT": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_INTERSECT_FILTER": {
                    "bedtools": "2.31.1"
                },
                "CUSTOM_GETCHROMSIZES": {
                    "getchromsizes": 1.2
                },
                "DEEPTOOLS_BAMCOVERAGE_MINUS": {
                    "deeptools": "3.5.1"
                },
                "DEEPTOOLS_BAMCOVERAGE_PLUS": {
                    "deeptools": "3.5.1"
                },
                "DRAGMAP_ALIGN": {
                    "dragmap": "1.2.1",
                    "samtools": "1.15.1",
                    "pigz": "2.3.4"
                },
                "DRAGMAP_HASHTABLE": {
                    "dragmap": "1.3.0"
                },
                "FASTP": {
                    "fastp": "0.23.4"
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GTF2BED": {
                    "perl": "5.26.2"
                },
                "HOMER_MAKETAGDIRECTORY": {
                    "homer": 4.11,
                    "samtools": 1.11
                },
                "PINTS_CALLER": {
                    "python": "3.10.6",
                    "pints": "1.1.8"
                },
                "PRESEQ_CCURVE": {
                    "preseq": "3.1.1"
                },
                "PRESEQ_LCEXTRAP": {
                    "preseq": "3.1.1"
                },
                "RSEQC_INFEREXPERIMENT": {
                    "rseqc": "5.0.2"
                },
                "RSEQC_READDISTRIBUTION": {
                    "rseqc": "5.0.2"
                },
                "RSEQC_READDUPLICATION": {
                    "rseqc": "5.0.2"
                },
                "SAMTOOLS_FLAGSTAT": {
                    "samtools": 1.2
                },
                "SAMTOOLS_IDXSTATS": {
                    "samtools": 1.2
                },
                "SAMTOOLS_INDEX": {
                    "samtools": 1.2
                },
                "SAMTOOLS_SORT": {
                    "samtools": 1.2
                },
                "SAMTOOLS_STATS": {
                    "samtools": 1.2
                },
                "SUBREAD_FEATURECOUNTS_GENE": {
                    "subread": "2.0.1"
                },
                "Workflow": {
                    "nf-core/nascent": "v2.3.0dev"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-12T13:24:45.988211144"
    }
}